moss/src/repository/manager.rs

```
@@ -184,16 +184,18 @@ fn update_meta_db(
     for payload in reader.payloads()? {
         // We only care about meta payloads for index files
         let Ok(stone::read::Payload::Meta(payload)) = payload else {
-                continue;
-            };
+            continue;
+        };
+
+        let metadata = package::Metadata::from_stone_payload(&payload)?;
+        // Create id from hash of meta
+        let id = package::Id::from(metadata.hash.clone().ok_or(Error::MissingHash)?);
```

Here's where we now construct the Id or error out if hash is missing. 

Now different consumers of meta-db can define their own requirements